Appc 1275 add events purchase intent to sdk

nunomonteiro opened the pull request:
**What does this PR do?**

Adds a purchase_intent event to the sdk that gets fired when user launches billing flow
Changes the package name
from: com.appcoins.sdk.android.billing
to: com.appcoins.billing.sdk

**Database changed?**

No

**Where should the reviewer start?**

- [ ] EventLogger.java

**How should this be manually tested?**

 Open bds-sdk proj and run toolbox. Press launch billing flow button.
Take note of the time you did this. 10 min later you can ask @jose.pereira to check if your event showed up. Pro tip: you can force the sku value to a very specific value to ease the verification process

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-1275](https://aptoide.atlassian.net/browse/APPC-1275)

**Questions:**
None

**Code Review Checklist**
- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

nunomonteiro committed:
Added EventLogger class to submit purchase event

Changed the application id to match the value expected by the BI platform (com.appcoins.billing.sdk instead of com.appcoins.sdk.android.billing)

nunomonteiro committed:
Fixed issues with firing purchase_intent event

brunovarelaAptoide commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/EventLogger.java:
```diff
@@ -0,0 +1,105 @@
+            int code = connection.getResponseCode();
+            System.out.println(code);
+
+            try(BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream(), "utf-8"))){
```
Code Style

brunovarelaAptoide replied:
Simplify the try. We have an IOException from the first try.

brunovarelaAptoide commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/EventLogger.java:
```diff
@@ -0,0 +1,105 @@
+        URL url = null;
+        String responseStr = "";
+
+        try {
```
Nested try catch

brunovarelaAptoide commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/EventLogger.java:
```diff
@@ -0,0 +1,105 @@
+
+    public static String ENTITY_NAME = "SDK";
+
+    private final String BASE_URL = "https://ws75.aptoide.com/api/7/";
```
Pass this URL to build config

nunomonteiro replied:
I like the way of thinking of this suggestion but I think that this var is only relevant to eventLogger

brunovarelaAptoide commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppCoinsAndroidBillingRepository.java:
```diff
@@ -96,12 +101,18 @@ public LaunchBillingFlowResult launchBillingFlow(String skuType, String sku, Str
     }
     try {
 
+      String valueStr = ""; //TODO how to get the purchase value, why do we need it?
```
Is this still in TODO?

brunovarelaAptoide commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppCoinsAndroidBillingRepository.java:
```diff
@@ -12,17 +12,22 @@
   public AppCoinsAndroidBillingRepository(int apiVersion, String packageName) {
     this.apiVersion = apiVersion;
     this.packageName = packageName;
+    this.eventLogger = new EventLogger();
```
Maybe create the EventLogger in CatapultBillingAppCoinsFactory and pass by parameter. We create all the Billing dependencies in CatapultBillingAppCoinsFactory.

nunomonteiro replied:
Awesome point!

nunomonteiro committed:
Changes from PR

Removed todo with value variable
Adopted code style
Removed nested try

nunomonteiro committed:
Architectural changes

Moved event logger to be passed by the factory and keep dependencies all in the same place
